vortex-io/src/object_store.rs

```diff
@@ -30,60 +26,22 @@ impl ObjectStoreReadAt {
 
 impl VortexReadAt for ObjectStoreReadAt {
     #[cfg_attr(feature = "tracing", tracing::instrument(skip(self)))]
-    fn read_byte_range(
-        &self,
-        pos: u64,
-        len: u64,
-    ) -> impl Future<Output = io::Result<Bytes>> + 'static {
+    async fn read_byte_range(&self, range: Range<u64>) -> io::Result<Bytes> {
         let object_store = self.object_store.clone();
         let location = self.location.clone();
-        Box::pin(async move {
```

Comment:
@a10y do you recall why we didn't just call get_range before?

The other thing is we were never using spawn_blocking, so we were just performing blocking I/O on the async thread?

Reply:
We wanted to use our own buffer, so we had to unwrap the get_range call. FWIW the underlying object store will spawn the file read_exact call

Reply:
But then we default to just returning the buffer which made it unnecessary to unwrap this, however, we didn't undo this change